pymeasure/instruments/mksinst/mks974b.py

```diff
@@ -0,0 +1,224 @@
+
+    setpoint_1 = Instrument.ChannelCreator(SetpointChannel, 1)
+
+    setpoint_2 = Instrument.ChannelCreator(SetpointChannel, 2)
+
+    setpoint_3 = Instrument.ChannelCreator(SetpointChannel, 3)
+
+    def __init__(self, adapter, name="MKS 974B vacuum pressure transducer", address=253, **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            address=address,
+            **kwargs
+        )
+
+    def id(self):
+        """ Get the identification of the instrument. """
+        return f"{self.manufacturer}{self.model} {self.device_type} ({self.serial_number})"
+
+    serial_number = Instrument.measurement(
+        "SN?", """Get the serial number of the instrument""",
+        cast=str,
+    )
+
+    hardware_version = Instrument.measurement(
+        "HV?", """Get the hardware version of the instrument""",
+        cast=str,
+    )
+
+    firmware_version = Instrument.measurement(
+        "FV?", """Get the firmware version of the instrument""",
+        cast=str,
+    )
+
+    device_type = Instrument.measurement(
+        "DT?", """Get the device type""",
+        cast=str,
+    )
+
+    manufacturer = Instrument.measurement(
+        "MF?", """Get the manufacturer name""",
+        cast=str,
+    )
+
+    model = Instrument.measurement(
+        "MD?", """Get the transducer model number""",
+        cast=str,
+    )
+
+    part_number = Instrument.measurement(
+        "MD?", """Get the transducer part number""",
+        cast=str,
+    )
+
+    operation_hours = Instrument.measurement(
+        "TIM?", """Get the operation hours of the instrument""",
+        cast=int,
+    )
+
+    temperature = Instrument.measurement(
+        "TEM?", """Get the MicroPirani sensor temperature""",
+    )
+
+    status = Channel.measurement(
+        "T?",
+        """Get transducer status""",
+        map_values=True,
+        values={"Ok": "O",
+                "MicroPirani failure": "M",
+                "Cold Cathode failure": "C",
+                "Piezo sensor failure": "Z",
+                "pressure dose setpoint exceeded": "R",
+                "Cold Cathode On": "G",
+                },
+    )
+
+    pirani_pressure = Instrument.measurement(
+        "PR1?", """Get MicroPirani sensor pressure""",
+    )
+
+    piezo_pressure = Instrument.measurement(
+        "PR2?", """Get Piezo differential sensor pressure""",
+    )
+
+    pressure = Instrument.measurement(
+        "PR4?", """Get combined pressure reading""",
+    )
+
+    coldcathode_pressure = Instrument.measurement(
+        "PR5?", """ Get Cold Cathode sensor pressure""",
+    )
+
+    unit = Instrument.control(
+        "U?", "U!%s",
+        """Control pressure unit used for all pressure readings from the instrument""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={"Torr": "TORR",
+                "mBar": "MBAR",
+                "Pascal": "PASCAL",
+                },
+        check_set_errors=True,
+    )
```

Comment:
You could use an enum for the values in order to make it easier to set the values.
At least you should mention the possible units in the docstring.
you could change the dictionary keys (and therefore possible user values) to the units abbreviation: Torr, mbar, Pa

Reply:
I now use an `Enum`. Since I did not see some hints in the documentation on how the Enum should be used I hope I found the intended way. Note that I introduced the use of Enum also in the MKS937B. This is a change with respect to v.0.13.1 so maybe needs to marked somehow?